src/Nethermind/Nethermind.Consensus/Tracing/GethStyleTracer.cs

```diff
@@ -167,6 +171,24 @@ public IEnumerable<string> TraceBlockToFile(Hash256 blockHash, GethTraceOptions
         return tracer.FileNames;
     }
 
+    public IEnumerable<string> TraceBadBlockToFile(Hash256 blockHash, GethTraceOptions options, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(blockHash);
+        ArgumentNullException.ThrowIfNull(options);
+
+        var block = _badBlockStore
+            .GetAll()
+            .Where(b => b.Hash == blockHash)
+            .FirstOrDefault()
```

Comment:
Maybe also require block number and don't load all bad blocks into memory?

Reply:
I feel you, but we can't, as this RPC method requires block hash only. Even if it didn't, it would be strange to require the user to provide both number and hash. RPC methods typically require either a number or a hash.